main.go

```diff
@@ -13,24 +13,71 @@ import (
 	goconfluence "github.com/virtomize/confluence-go-api"
 )
 
+type Article struct {
+	ID             string
+	Title          string
+	LastUpdateTime time.Time
+}
+
 func getDomainName(confluenceURL string) string {
-	url, err := url.Parse(confluenceURL)
+	parsedURL, err := url.Parse(confluenceURL)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	return url.Scheme + "://" + url.Host
+	return parsedURL.Scheme + "://" + parsedURL.Host
+}
+
+func getOldArticles(api *goconfluence.API, pageID string, now time.Time, maxDepth int, ageThresholdHours float64) ([]Article, error) {
+	var oldArticles []Article
+	err := getOldArticlesRecursive(api, pageID, now, maxDepth, 0, ageThresholdHours, &oldArticles)
+	return oldArticles, err
+}
+
+func getOldArticlesRecursive(api *goconfluence.API, pageID string, now time.Time, maxDepth, currentDepth int, ageThresholdHours float64, oldArticles *[]Article) error {
+	if currentDepth > maxDepth {
+		return nil
+	}
+
+	childPages, err := api.GetChildPages(pageID)
+	if err != nil {
+		return fmt.Errorf("error getting child pages: %w", err)
+	}
+
+	for _, page := range childPages.Results {
+		history, err := api.GetHistory(page.ID)
+		if err != nil {
+			return fmt.Errorf("error getting history: %w", err)
+		}
```

Comment:
What do you think about extending the error handling with the specific pageID the error occurs, something like

```go
        if err != nil {
            return fmt.Errorf("failed to fetch history for page %s: %w", page.ID, err)
        }
```
But I am not sure if this would be super helpful in some cases (but maybe for some permission issues on subpages)